### PR TITLE
Airspeed scaling

### PR DESCRIPTION
**Pull request by bresch**
Following the discussion that started [here](https://github.com/PX4/ecl/pull/408#issuecomment-381969477), I finally decided to write down the equations behind the airspeed scaling strategy present in the FW rate controller. We should also update the diagram and add the the indicated airspeed signal to the PI_Scaling block (next to the true airspeed signal V_T). @hamishwillee Where can I update the schematic?

We should also update the firmware as it only uses true airspeed scaling for both PI and FF outputs.

Here is a preview of this PR:
https://bresch.gitbooks.io/devguide/content/en/flight_stack/controller_diagrams.html

FYI @tstastny , @RomanBapst , @priseborough 
If you see any mistake, please tell me :)

Tasks:
- [x] Add IAS to the diagram and split scaling block
- [x] Add related parameters and how to set them
-  Modify the firmware to use both IAS and TAS scaling

### CODE REVIEW

File: en/flight_stack/controller_diagrams.md [hunk 1]
```diff
@@ -33,3 +33,83 @@ The angular position of the control effectors (ailerons, elevators, rudders, ...
...
 
 The roll and pitch controllers have the same structure and the longitudinal and lateral dynamics are assumed to be uncoupled enough to work independently. The yaw controller, however, generates its yaw rate setpoint using the turn coordination constraint in order to minimize lateral acceleration, generated when the aircraft is slipping. The yaw rate controller also helps to counteract adverse yaw effects (https://youtu.be/sNV_SDDxuWk) and to damp the [Dutch roll mode](https://en.wikipedia.org/wiki/Dutch_roll) by providing extra directional damping.
+
+### Airspeed scaling
+The objective of this section is explain with the help of equations why and how the output of the rate PI and feedforward (FF) controllers can be scaled with airspeed to improve the control performance. We will first present the simplified linear dimentional moment equation on the roll axis, then show the influence of airspeed on the direct moment generation and finally, the influence of airspeed during a constant roll.
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md]: - is TO explain
- linear dimentional -> dimensional

File: en/flight_stack/controller_diagrams.md [hunk 2]
```diff
@@ -33,3 +33,83 @@ The angular position of the control effectors (ailerons, elevators, rudders, ...
...
+
+As shown in the fixed-wing attitude controller above, the rate controllers produce angular acceleration setpoints for the control allocator (here named "mixer"). In order to genereate these desired angular accelerations, the mixer produces torques using available aerodynamic control surfaces (e.g.: a standard airplane typically has two ailerons, two elevators and a rudder). The torques generated by those control surfaces is highly influenced by the relative airspeed and the air density, or more precisely, by the dynamic pressure. If no airspeed scaling is made, a controller tightly tuned for a certain cruise airspeed will make the aircraft oscillate at higher airspeed or will give bad tracking performance at low airspeed.
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md]: In order to genereate -> generate

File: en/flight_stack/controller_diagrams.md [hunk 3]
```diff
@@ -33,3 +33,83 @@ The angular position of the control effectors (ailerons, elevators, rudders, ...
...
+
+The reader should be aware of the difference between the [true airspeed (TAS)](https://en.wikipedia.org/wiki/True_airspeed) and the [indicated airspeed (IAS)](https://en.wikipedia.org/wiki/Indicated_airspeed) as their values are significatively different when not flying at sea level.
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md]: are significatively -> significantly

File: en/flight_stack/controller_diagrams.md [hunk 4]
```diff
@@ -33,3 +33,83 @@ The angular position of the control effectors (ailerons, elevators, rudders, ...
 The feedforward gain is used to compensate for aerodynamic damping. Basically, the two main components of body-axis moments on an aircraft are produced by the control surfaces (ailerons, elevators, rudders, - producing the motion) and the aerodynamic damping (proportional to the body rates - counteracting the motion). In order to keep a constant rate, this damping can be compensated using feedforward in the rate loop.
 
 The roll and pitch controllers have the same structure and the longitudinal and lateral dynamics are assumed to be uncoupled enough to work independently. The yaw controller, however, generates its yaw rate setpoint using the turn coordination constraint in order to minimize lateral acceleration, generated when the aircraft is slipping. The yaw rate controller also helps to counteract adverse yaw effects (https://youtu.be/sNV_SDDxuWk) and to damp the [Dutch roll mode](https://en.wikipedia.org/wiki/Dutch_roll) by providing extra directional damping.
+
+### Airspeed scaling
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md]: To "Airspeed Scaling"

File: en/flight_stack/controller_diagrams.md (hunk 2 above)
Comment (hamishwillee) [reply en/flight_stack/controller_diagrams.md]: FYI, IMO above is a really good summary of why the discussion is relevant.

File: en/flight_stack/controller_diagrams.md [hunk 5]
```diff
@@ -33,3 +33,83 @@ The angular position of the control effectors (ailerons, elevators, rudders, ...
...
+
+where $$\beta$$ is the sideslip angle, $$p$$ the body roll rate and $$\delta_a$$ the aileron deflection.
+
+Assuming a symmetric ($$C_{\ell_0} = 0$$) and coordinated ($$\beta = 0$$) aircraft, the equation can be simplified using only the rollrate damping and the roll moment produced by the ailerons
+
+$$\ell = \frac{1}{2}\rho V_T^2 S b \left [C_{\ell_{\delta_a}} \:\delta_a + C_{\ell_p}\:\frac{b}{2V_T} \: p \right ]$$.
+
+This final equation is then taken as a baseline for the two next subsections to determine the airspeed scaling expression required for the PI and the FF controllers.
+
+#### a.Torque (PI) scaling
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md]: Do we need the "a" and "b" in headings?

File: en/flight_stack/controller_diagrams.md [hunk 6]
```diff
@@ -33,3 +33,83 @@ The angular position of the control effectors (ailerons, elevators, rudders, ...
...
+
+Rearranging to extract the ideal ailerons deflection gives
+
+$$\delta_a = -\frac{b \: C_{\ell_p}}{2 \: C_{\ell_{\delta_a}}} \frac{1}{V_T} \: p$$.
+
+The first fraction gives the value of the ideal feedforward and we can see that the scaling is linear to the TAS. 
+Note that the negative sign is then absorbed by the roll damping derivative which is also negative.
+
+#### Conclusion
+The output of the rate PI controller has to be scaled with the indicated airspeed (IAS) squared and the output of the rate feedforward (FF) has to be scaled with the true airspeed (TAS)
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md:111]: So presumably we have scaling parameters for tuning that a user can specify for these. Can we link to the names and where we suggest some reasonable starting points for setting these?

**Comment by hamishwillee** [comment]
>  @hamishwillee Where can I update the schematic?

To make it easy, the URL is in a comment in the source, below the image - ie [in this case](
https://github.com/PX4/Devguide/pull/555/files#diff-808b15f386ee097787276cc418f31195R24)

This looks good. A couple of typos inline.

@tstastny , @RomanBapst , @priseborough
Can you please give technical review. It looks sane for me, but I don't have the maths to be sure nothing has been missed and that is all follows.

File: en/flight_stack/controller_diagrams.md (hunk 6 above)
Comment (bresch) [reply en/flight_stack/controller_diagrams.md:111]: The good thing about this scaling is that it doesn't require to tune a parameter. One just has to tune the attitude controller at the cruise airspeed and set this cruise airspeed correctly. I will add that to the documentation, thanks!

**Comment by bresch** [comment]
@hamishwillee Thanks a lot for your review, I fixed the typos and I'll make the changes in the diagram.

**Comment by hamishwillee** [comment]
@bresch You are most welcome. Just let me know when you are happy for it to be merged.

**Comment by hamishwillee** [comment]
Looks good. I added a subedit comment. Nearly done - let me know when you want to merge.

**Comment by bresch** [comment]
@hamishwillee I'm finally quite happy with it. I think it's ready to be merged and we can still modify it later if needed.
I'll do the according modifications in the Firmware asap.

**Comment by hamishwillee** [comment]
Yes, it is excellent. Thank you. Merged.

**Pull request merged**